test-case/check-audio-equalizer.sh

```diff
@@ -64,30 +67,41 @@ func_test_eq()
 }
 
 # this function performs IIR/FIR filter test
-# param1 must be must be iir or fir
+# param1 must be must be component name
 func_test_filter()
 {
     local testfilter=$1
     dlogi "Get amixer control id for $testfilter"
-    # TODO: Need to match alsa control id with the filter in the pipeline,
-    #       currently the test discards EQ pipelines except first one.
-    Filterid=$(amixer -D hw:"$sofcard" controls | sed -n -e "/eq${testfilter}/I "'s/numid=\([0-9]*\),.*/\1/p' | head -1)
+    Filterid=`$my_dir/../tools/topo_effect_kcontrols.py $tplg $testfilter`
     if [ -z "$Filterid" ]; then
         die "can't find $testfilter"
     fi
 
-    declare -a FilterList=($(ls -d "${my_dir}"/eqctl/eq_"${testfilter}"_*.txt))
-    nFilterList=${#FilterList[@]}
-    dlogi "$testfilter list, num= $nFilterList, coeff files= ${FilterList[*]}"
+    if is_ipc4; then
+        ipc_dir="ipc4"
+    else
+        ipc_dir="ipc3"
+    fi
+
+    if [[ ${Filterid^^} == *"IIR"* ]]; then
+        comp_dir="eq_iir"
+    elif [[ ${Filterid^^} == *"FIR"* ]]; then
+        comp_dir="eq_fir"
+    else
+        die "Not supported control: $Filterid"
+    fi
+
+    nFilterList=$(ls -1 "${my_dir}"/eqctl/$ipc_dir/$comp_dir/*.txt | wc -l)
```

Comment:
This is enough to please shellcheck:
```suggestion
    nFilterList=$(find "${my_dir}/eqctl/$ipc_dir/$comp_dir/" -name '*.txt' | wc -l)
```

Reply:
Yep